sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/BoundedReadFromUnboundedSource.java

```diff
@@ -107,6 +108,21 @@ public String getKindString() {
     return "Read(" + approximateSimpleName(source.getClass()) + ")";
   }
 
+  @Override
+  public void populateDisplayData(DisplayData.Builder builder) {
+    builder
+        .add("source", source.getClass())
+        .include(source);
+
+    if (maxNumRecords != Long.MAX_VALUE) {
+      builder.add("maxRecords", maxNumRecords);
```

Comment:
don't drop the `Num`?

Do you prefer English or Java names for keys. Could be "Maximum number of records returned", or "maxNumRecords".